pkg/kubelet/kubelet.go

```diff
@@ -1011,6 +1034,24 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 		if err != nil {
 			return err
 		}
+
+		//
+		// TODO: call site here is probably not ideal, but this seems like a good place
+		// in this first iteration to determine whether preconditions are satisfied.
+		//
+		// I considered making the precondition an attribute of a container, but I don't
+		// like the consequences of that factoring, as it would be possible for:
+		//
+		// 1.  Pods to be incompletely started
+		// 2.  Resources (ex, network container; host volumes) to be consumed by pods that
+		//     are not ready to start
+		//
+		err = kl.checkPodPreconditions(pod)
+		if err != nil {
+			glog.Errorf("Pod has failed preconditions: %v; Skipping pod %q", err, podFullName)
```

Comment:
Assuming we go forward with this approach:

This can't log an error. It needs to be at a high verbosity level. Even that won't be useful, however. Instead, we should create a PodCondition entry for this case and add LastProbeTime, LastTransitionTime, Reason, and Message to PodCondition.